coremltools/models/neural_network/flexible_shape_utils.py

```diff
@@ -712,18 +715,23 @@ def add_multiarray_ndshape_enumeration(spec, feature_name, enumerated_shapes):
 
     eshape_len = len(feature.type.multiArrayType.enumeratedShapes.shapes)
 
+    shapes_added_so_far = []
```

Comment:
Not fully follow the usage of this `shapes_added_so_far`. Is this list of shapes? What's the usage?

Reply:
Nevermind, saw that now. Why don't we use a `set` in this case?

Reply:
The elements are of type lists, and set of lists are not allowed.

Reply:
I see, tuple should work if we make shape a tuple.